nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2587 @@
+
+    return collection
+
+
+def _download_image_nii_file(image_info, collection, download_params):
+    """Download an image (.nii.gz) file from Neurovault.
+
+    Parameters
+    ----------
+    image_info : dict
+        Image metadata.
+
+    collection : dict
+        Corresponding collection metadata.
+
+    download_params : dict
+       General information about download session, containing e.g. the
+       data directory (see `_read_download_params` and
+       `_prepare_download_params for details`)
+
+    Returns
+    -------
+    image_info : dict
+        Image metadata with local paths added to it.
+
+    collection : dict
+        Corresponding collection metadata with local paths added to it.
+
+    """
+    image_info = image_info.copy()
+    image_id = image_info['id']
+    image_url = image_info['file']
+    image_file_name = 'image_{0}.nii.gz'.format(image_id)
+    image_relative_path = os.path.join(
+        collection['relative_path'], image_file_name)
+    image_absolute_path = os.path.join(
+        collection['absolute_path'], image_file_name)
+    _simple_download(
+        image_url, image_absolute_path,
+        download_params['temp_dir'], verbose=download_params['verbose'])
+    image_info['absolute_path'] = image_absolute_path
+    image_info['relative_path'] = image_relative_path
+    return image_info, collection
+
+
+def _download_image_reduced_rep(image_info, collection, download_params):
+    """Download reduced representation (.npy) of an image.
+
+    Parameters
+    ----------
+    image_info : dict
+        Image metadata.
+
+    collection : dict
+        Corresponding collection metadata.
+
+    download_params : dict
+       General information about download session, containing e.g. the
+       data directory (see `_read_download_params` and
+       `_prepare_download_params for details`)
+
+    Returns
+    -------
+    image_info : dict
+        Image metadata with reduced representation path added to it.
+
+    collection : dict
+        Corresponding collection metadata.
+
+    """
+    reduced_image_url = image_info.get('reduced_representation')
+    if not download_params['fetch_reduced_rep'] or reduced_image_url is None:
+        return image_info, collection
+
+    reduced_image_name = 'image_{0}_reduced_rep.npy'.format(image_info['id'])
+    reduced_image_relative_path = os.path.join(
+        collection['relative_path'], reduced_image_name)
+    reduced_image_absolute_path = os.path.join(
+        collection['absolute_path'], reduced_image_name)
+    _simple_download(
+        reduced_image_url, reduced_image_absolute_path,
+        download_params['temp_dir'], verbose=download_params['verbose'])
+    image_info = image_info.copy()
+    image_info['reduced_representation'
+               '_relative_path'] = reduced_image_relative_path
+    image_info['reduced_representation'
+               '_absolute_path'] = reduced_image_absolute_path
```

Comment:
We should remove all the code related to reduced representation, it will make things simple.